tests/Functional/SalesDataProviderTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Tests\Functional;
+
+use Fidry\AliceDataFixtures\LoaderInterface;
+use Fidry\AliceDataFixtures\Persistence\PurgeMode;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Sylius\Component\Core\Dashboard\SalesDataProviderInterface;
+use Sylius\Component\Core\Dashboard\SalesSummary;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Order\Processor\OrderProcessorInterface;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\BrowserKit\Client;
+
+final class SalesDataProviderTest extends WebTestCase
+{
+    /** @var Client */
+    private static $client;
+
+    /** @test */
+    public function it_provides_simple_year_sales_summary_grouped_per_month(): void
+    {
+        $salesSummary = $this->getSummaryForChannel('CHANNEL');
+
+        $this->assertInstanceOf(SalesSummary::class, $salesSummary);
+        $this->assertSame(
+            [(new \DateTime('-1 month'))->format('m.y'), (new \DateTime('-2 month'))->format('m.y'), (new \DateTime('-3 month'))->format('m.y')],
+            $salesSummary->getMonths()
+        );
+        $this->assertSame(['20.00', '20.00', '30.00'], $salesSummary->getSales());
+    }
+
+    /** @test */
+    public function it_provides_different_data_for_each_channel_and_only_paid_orders(): void
+    {
+        $salesSummary = $this->getSummaryForChannel('EXPENSIVE_CHANNEL');
+
+        $this->assertInstanceOf(SalesSummary::class, $salesSummary);
+        $this->assertSame(
+            [(new \DateTime('-1 month'))->format('m.y')],
+            $salesSummary->getMonths()
+        );
+        $this->assertSame(['330.00'], $salesSummary->getSales());
+    }
+
+    protected function setUp(): void
```

Comment:
Could you move it all the way up? It's protected, but in the end, it executes before the tests themselves, so it makes more sense to read it first as well.